### DISCUSSION TITLE
Shall we keep the parameterless AddHttpCorrelation method

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.webapi#319] [General]
We have made the parameterless `AddHttpCorrelation` method obsolete in favor of the `AddHttpCorrelation` method which takes a `HttpCorrelationInfoOptions` object as parameter.

However, I've seen that, when you want to use the default implementation, you now have to call this method like this:

```csharp
builder.AddHttpCorrelation((HttpCorrelationInfoOptions options) => { });
```

I think this is a bit cumbersome and pollutes the code.
If you want to stick with the defaults, why not just leave it as is and let people call this just like:

```csharp
builder.AddHttpCorrelation();
```

This is simple and straightforward.
We can keep the parameterless `AddHttpCorrelation` overload.   This parameterless overload can be implemented like this:

```csharp
builder.AddHttpCorrelation((HttpCorrelationInfoOptions options) => { });
```

First mentionned [here](https://github.com/arcus-azure/arcus.templates/pull/559/files#r851167202)

**Comment by stijnmoreels** [comment]
This is part of an ongoing feature where we will eventually remove the parameterless extension #291 and replace it with a new one with the HTTP-specific options #293.

**Comment by stijnmoreels** [comment]
I'll move this to an discussion, as we already have issues for this.
